[android] Reworked descriptions in the editor

map-per opened the pull request:
Reworked descriptions in the editor:

- Placed info about OSM at the top to prevent people from using the editor for personal notes. Moreover it is a good practice for a privacy focused app to inform users about what and where data is send.
- Improved the hint of the the note field. Currently the description below the field is very misleading and many people use it like a changeset comment as they don't know it's for OSM notes.
- Apart from the slight change in the "Note to OpenStreetMap volunteers" string this PR doesn't affect iOS

| Now | Before |
|--------|--------|
|![%2Fstorage%2Femulated%2F0%2FDCIM%2FScreenshots%2FIMG_20240824_210007](https://github.com/user-attachments/assets/49ff54ea-a773-4a4d-afe2-0670aa52603e)|![Screenshot_2024-08-18-18-17-28-581_app organicmaps](https://github.com/user-attachments/assets/f69e5452-724c-4d9f-8607-9df56bfc839b)|
|![Screenshot_2024-08-24-20-45-21-816_app organicmaps debug](https://github.com/user-attachments/assets/58defcfd-8594-40de-a6de-7c8c5d8d7bdc)|![Screenshot_2024-08-18-18-15-01-604_app organicmaps](https://github.com/user-attachments/assets/da08b60e-8f24-4df8-9c65-4469207812e3)|

### Todo:
- [x] Translations

biodranik commented on android/app/src/main/res/layout/fragment_editor.xml:
```diff
@@ -284,15 +283,17 @@
             android:layout_height="wrap_content"
             android:layout_marginTop="@dimen/margin_half"
-            android:text="Your note will be published on OSM. Please describe errors or additional details that cannot be edited in Organic Maps, so other OSM volunteers can help to solve the problem."
+            android:text="Use this to start a discussion with other OSM volunteers. Please describe errors or additional details that cannot be edited in Organic Maps."
```
Is this text replaced by another from strings.txt?

map-per replied:
I now added `editor_note_description` to strings.txt

biodranik commented on data/strings/strings.txt:
```diff
@@ -14973,44 +15016,45 @@
-    vi = Những thay đổi đề nghị của bạn sẽ được gửi đến cộng đồng OpenStreetMap. Mô tả các chi tiết không thể sửa trong Organic Maps.
-    zh-Hans = 您建议的更改将发送至 OpenStreetMap 社区。说明无法在 Organic Maps 编辑的详情。
-    zh-Hant = 您建議的變更將傳送至 OpenStreetMap 社群。說明無法在 Organic Maps 中編輯的詳細資料。
+    en = Your edits are uploaded to the public OpenStreetMap (OSM) database, so don't add personal or copyrighted information.
```
What do you think about this?

    en = Your edits are uploaded to the public OpenStreetMap.org (OSM) database. Please do not add personal or copyrighted information.

Jean-BaptisteC replied:
I think we can remove public in sentence

biodranik replied:
No, it's critically important, based on our experience.

map-per replied:
I check all OM edits with a boundary box intersecting munich and there are regularly people who upload personal notes

map-per replied:
I updated the description

biodranik commented on data/strings/strings.txt:
```diff
@@ -14889,46 +14889,89 @@
-    vi = Gửi ghi chú cho ban biên tập OSM
-    zh-Hans = 给OSM编辑器发送标记
-    zh-Hant = 給OSM編輯器發送標記
+    en = Open a discussion note
```
?
```
    en = Open a discussion note at OpenStreetMap.org
```

map-per replied:
I think with the description below `OpenStreetMap.org` would be rather redundant

biodranik commented on android/app/src/main/java/app/organicmaps/editor/EditorFragment.java:
```diff
@@ -466,7 +466,7 @@ private void initViews(View view)
     mOpeningHours.setOnClickListener(this);
     final View cardMore = view.findViewById(R.id.cv__more);
     mDescription = findInput(cardMore);
-    cardMore.findViewById(R.id.about_osm).setOnClickListener(this);
```
Is cardMore used somewhere? Can it be deleted?

map-per replied:
It's used in line 468 for finding the mDescription

biodranik reviewed (commented):
Thanks! The next step would be to make it simpler (if possible).

Now the note has 3 texts:
1. Title
2. Description
3. Edit text hint

Can we squash them and make a better wording text, and use only edit text hint, without title and description? OM is about simplicity and efficiency ;-)

map-per commented:
> Thanks! The next step would be to make it simpler (if possible).
> 
> Now the note has 3 texts:
> 
>     1. Title
> 
>     2. Description
> 
>     3. Edit text hint
> 
> 
> Can we squash them and make a better wording text, and use only edit text hint, without title and description? OM is about simplicity and efficiency ;-) Maybe in a separate PR?

- I would definitely keep the title to make the card view layout consistent with the others
- Google discourages using multi line lable texts (see: https://m2.material.io/components/text-fields#anatomy) (and they probably don't look good). So we could only use placeholder text, but that would disappear as soon as the user starts typing, which is not good in my opinion.

I think the new Notes card view is good as it is now

biodranik commented:
1. An important link to OSM disappeared. Remember, more than 90% of OM users NEVER heard about OSM and do not understand where map data comes from.
2. There is no reason to keep 3 different texts for the note. Smaller text is usually skipped. Try to merge at least the title and description, make it succinct, and leave some other important piece of information in the edit box hint.

map-per commented:
Ok, then how about "Open a discussion note at OpenStreetMap.org" as the title and "Detailed description of things that cannot be edited in Organic Maps" as placeholder text?

biodranik commented:
That sounds better! How will it look?

map-per commented:
I don't know, heaven't implemented it yet 😃

map-per commented:
This is how it looks:

| | |
|--------|--------|
| ![signal-2023-10-16-085539_002](https://github.com/organicmaps/organicmaps/assets/79519062/a9182fb8-1fd6-4f6d-b649-bd11e31cb27e)| ![signal-2023-10-16-085539_003](https://github.com/organicmaps/organicmaps/assets/79519062/817e4d36-6085-4e2a-bbee-f835b332c2b4)|

biodranik commented:
1. Can "More about OpenStreetMap.org" be integrated into a link in the text above?
2. I'm a new user. I don't know what does that "discussion" mean. I'm not sure yet what osm means. Wouldn't it be clearer to explicitly mention that the note will be checked by other community volunteers? The goal of this field here is to improve the map, not to generate tons of noisy notes for the OpenStreetMap.org community. 
3. OM can't edit everything. That's another reason for this field. Does it make sense to add a link to "edit it using any other editor"? We can make a stub page on our website and describe popular OpenStreetMap.org editors.

map-per commented:
> 2. I'm a new user. I don't know what does that "discussion" mean. I'm not sure yet what osm means. Wouldn't it be clearer to explicitly mention that the note will be checked by other community volunteers? The goal of this field here is to improve the map, not to generate tons of noisy notes for the OpenStreetMap.org community.

You wanted a shorter description.

Can we do further improvements in a separate PR?

map-per commented:
> 1. Can "More about OpenStreetMap.org" be integrated into a link in the text above?

It's possible for sure, but I don't now how that works

> I'm a new user. I don't know what does that "discussion" mean. I'm not sure yet what osm means. Wouldn't it be clearer to explicitly mention that the note will be checked by other community volunteers? The goal of this field here is to improve the map, not to generate tons of noisy notes for the OpenStreetMap.org community.

I hope most people will know what a discussion is (that's not an OSM specific term)

> OM can't edit everything. That's another reason for this field. Does it make sense to add a link to "edit it using any other editor"? We can make a stub page on our website and describe popular OpenStreetMap.org editors.

That's a good Idea, so far I couldn't find a really good external page to link to. I would place that link in the `editor_detailed_description` text at the top.

map-per commented:
The new website is rather a longterm project or? As private bookmarks uploaded to OSM are a bigger OrganicMaps problem it would be nice to have this initial fix in the next release.
For me this is rather exhausting if just changing some descriptions gests so complicated.

biodranik commented:
Sorry for the slow reviews, we're very overloaded with many other tasks ATM. 

> As private bookmarks uploaded to OSM are a bigger OrganicMaps problem it would be nice to have this initial fix in the next release.

Do you have any data/numbers about the number of bad changesets of OM Android clients you're referring to as a "bigger" issue?

map-per commented:
I checked all OrgancMaps and Maps.me changesets with a boudary box intersecting munich of the last year once again: 
In 277 OM changesets there was one edit with a private bookmark and in 107 Maps.me edits there were 3. These are edits that are definitely private bookmakrs as they contain names like "HOME" or the names of the users friends. There might possible be more unnamed private bookmarks which can't be spotted without local knowledge or can not be distinguished form vandalism. (an explanation for some vandalism cases could be that user don't know that a long tap selects any point, so they add a POI first to select the place with a short tap)

So this doesn't happen extremly often but every once and again.

biodranik commented:
I've tested and reviewed updated dialogs. Moving the important note about OSM at the top is okay and can be merged.

Regarding the note, it's more complex. Firstly, some translations are incorrect. Secondly, many people won't understand why some discussion should be opened when they edit the map, and what this discussion means. This note field was introduced with the main goal of adding some changes to the map that can not be done using OM. This is _not clear_ with the proposed text, the previous text was _clearer_ for that goal IMO (but not the best either).

> Currently many people use the field like a changeset comment.

What are the stats? Looks like the better text should 1) let people know that it's not a comment on their changes and 2) clearly specify that this field should be filled _only_ if users can't add/edit something using OM.

map-per commented:
Do you have any suggestions for a good text?

How about "Ask other OpenStreetMap volunteers for help if you can't edit something with Organic Maps."?

map-per commented:
And here the stats:
Out of the last 25 notes in Munich 6 notes didn't contain further information, but a changeset comment

biodranik commented:
Before, the text with the warning about OSM-uploaded changes was trying to solve both:
1. Notify about the publicly uploaded changes.
2. Ask users to add comments if something can't be edited.

Splitting the text is a good idea, but always wasting a vertical space for the warning at the top may be an overkill for experienced editors (that's why it was done at the bottom initially, to have more useful info on the screen). We may consider to hide it later.

Regarding the comment/note field, two important moments:
1. It is absent on iOS when adding a feature (before OM, the UX was always iOS-first). Should it really be displayed when something is added?
2. As I wrote to you before, the best UI for the note would be to remove all labels and leave only the empty editable text field with the hint like "Optionally, describe any additional details that can not be added or edited in Organic Maps".

map-per commented:
> 1. It is absent on iOS when adding a feature (before OM, the UX was always iOS-first). Should it really be displayed when something is added?

I think that doesn't really matter, both is fine. But it shoud, of course, be consistent in Android and ios. I would exclude that from this PR.


> As I wrote to you before, the best UI for the note would be to remove all labels and leave only the empty editable text field with the hint like "Optionally, describe any additional details that can not be added or edited in Organic Maps".

In my opinion removing the heading is bad UI as this is inconsistent with the other name, address and details sections and you end up with a text field without any heading or description as soon as you start typing. I strongly object to the removal. Would you also accept a PR that keeps the heading?

For the hint text I prefere "Ask other OpenStreetMap volunteers for help if you can't edit something with Organic Maps. (optional)". As you said, mentioning community volunteers is better.

map-per commented:
Updated the string:

| Header | Header |
|--------|--------|
|![Screenshot_2023-10-31-20-03-59-112_app organicmaps beta](https://github.com/organicmaps/organicmaps/assets/79519062/ab543a99-b750-4a3e-b8f8-c219e194a3f3)|![Screenshot_2023-10-31-20-04-22-407_app organicmaps beta](https://github.com/organicmaps/organicmaps/assets/79519062/0a6a25af-b621-4324-b300-a68b7c230808)|

Jean-BaptisteC commented on android/app/src/main/res/layout/fragment_editor.xml:
```diff
@@ -13,6 +13,22 @@
+    <TextView
+      android:layout_width="match_parent"
+      android:layout_height="wrap_content"
+      android:layout_marginTop="@dimen/margin_half"
```
```suggestion
      android:layout_marginTop="@dimen/margin_base"
```

Jean-BaptisteC commented on android/app/src/main/res/layout/fragment_editor.xml:
```diff
@@ -13,6 +13,22 @@
+      android:textAppearance="@style/MwmTextAppearance.Body4"/>
+    <TextView
+      android:id="@+id/about_osm"
+      android:layout_width="match_parent"
```
```suggestion
      android:layout_width="wrap_content"
```

Jean-BaptisteC reviewed (approved):
Some improvements for layout

biodranik commented:
Thanks! Looks better, but texts are too long to read. "More about OpenStreetMap" can be removed, and OpenStreetMap.org can be highlighted instead.

The bottom one is harder... Let's try this:
- Replace the title with the hint
- Leave (optional) in the hint

A more succinct phrase is needed...

map-per commented:
Ok, what strings should I use for title and hint? If everything I suggest is either to long or to short you have to come up with something yourself.

map-per commented:
Let's dig this PR up again. How about:

Moving the link to OSM into this sentence at the top:
> "Your edits are uploaded to the public [OpenStreetMap](https://wiki.openstreetmap.org/wiki/About_OpenStreetMap) database. Please do not add personal or copyrighted information."

Keep the heading similar to the current "Send a note to OSM editors" string but write out OSM and replace editors with volunteers:
> "Send a note to OpenStreetMap volunteers"

Keep the hint similar to the second sentence of the current text:
> "Describe map errors or additional details that cannot be edited with Organic Maps (optional)"


This way the content is not changed very much compared to the current version, but the note section and the mentioning OSM section are disentangled to prevent confusion.

biodranik commented:
Texts are more or less ok, but taking important vertical space at the top, without a way to hide it, always, even when users already learned it by heart, is not an OM way.

map-per commented:
I finally found time to work on OM again and did the changes as suggested in May.

The updated description with screenshots can be found in the first post (https://github.com/organicmaps/organicmaps/pull/6173#issue-1920818726).

biodranik commented on data/strings/strings.txt:
```diff
@@ -16542,8 +16542,16 @@
 
+  [editor_note_hint]
+    tags = android
+    en = Describe map errors or additional details that cannot be edited with Organic Maps (optional)
```
1. Maybe remove `(optional)` from here and write `Optional note to OpenStreetMap volunteers` in the title?
2. What about a shorter version?
```
    en = Describe errors on the map or things that cannot be edited with Organic Maps
    af = Beskryf foute in die kaart of wat nie met Organiese Kaarte geredigeer kan word nie
    ar = وصف الأخطاء على الخريطة أو الأشياء التي لا يمكن تحريرها باستخدام الخرائط العضوية
    az = Xəritədəki səhvləri və ya Üzvi Xəritələrdən istifadə edərək redaktə edilə bilməyənləri təsvir edin
    be = Апішыце памылкі на карце ці тое, што нельга рэдагаваць з дапамогай Organic Maps
    bg = Описване на грешки в картата или неща, които не могат да бъдат редактирани с Organic Maps
    ca = Descriu errors al mapa o què no es pot editar amb Mapes orgànics
    cs = Popište chyby na mapě nebo věci, které nelze upravit pomocí aplikace Organic Maps.
    da = Beskriv fejl på kortet eller ting, der ikke kan redigeres med Organic Maps
    de = Beschreiben Sie Fehler auf der Karte oder Dinge, die nicht mit Organic Maps bearbeitet werden können
    el = Περιγράψτε λάθη στο χάρτη ή πράγματα που δεν μπορούν να επεξεργαστούν με τους Organic Maps
    es = Describa los errores en el mapa o las cosas que no se pueden editar con Organic Maps
    et = Kirjeldage kaardil olevaid vigu või asju, mida ei saa Organic Mapsiga muuta.
    eu = Deskribatu mapan akatsak edo mapa organikoak erabiliz editatu ezin dena
    fa = خطاهای موجود در نقشه یا مواردی که با استفاده از نقشه های ارگانیک قابل ویرایش نیستند را شرح دهید
    fi = Kuvaile kartassa olevia virheitä tai asioita, joita ei voi muokata Organic Mapsilla.
    fr = Décrivez les erreurs sur la carte ou les éléments qui ne peuvent pas être modifiés avec Organic Maps.
    he = תאר שגיאות במפה או מה שלא ניתן לערוך באמצעות מפות אורגניות
    hi = मानचित्र में त्रुटियों का वर्णन करें या ऑर्गेनिक मानचित्र का उपयोग करके क्या संपादित नहीं किया जा सकता है
    hu = Írja le a térképen található hibákat vagy olyan dolgokat, amelyeket nem lehet az Organic Maps segítségével szerkeszteni.
    id = Jelaskan kesalahan pada peta atau hal-hal yang tidak dapat diedit dengan Peta Organik
    it = Descriva gli errori sulla mappa o le cose che non possono essere modificate con Organic Maps
    ja = 地図上のエラーやOrganic Mapsで編集できないものについて説明します。
    ko = 지도의 오류 또는 오가닉 맵으로 편집할 수 없는 항목에 대해 설명합니다.
    lt = Aprašykite žemėlapyje esančias klaidas arba dalykus, kurių negalima redaguoti naudojant "Organic Maps
    mr = नकाशामधील त्रुटी किंवा सेंद्रिय नकाशे वापरून काय संपादित केले जाऊ शकत नाही याचे वर्णन करा
    nb = Beskriv feil på kartet eller ting som ikke kan redigeres med Organic Maps
    nl = Beschrijf fouten op de kaart of dingen die niet bewerkt kunnen worden met Organic Maps
    pl = Proszę opisać błędy na mapie lub rzeczy, których nie można edytować za pomocą Organic Maps.
    pt = Descreva erros no mapa ou coisas que não podem ser editadas com o Organic Maps
    ro = Descrieți erorile de pe hartă sau lucrurile care nu pot fi editate cu Organic Maps
    ru = Опишите ошибки на карте или то, что нельзя редактировать с помощью Organic Maps
    sk = Popíšte chyby na mape alebo veci, ktoré sa nedajú upraviť pomocou aplikácie Mapy Organic
    sv = Beskriv fel på kartan eller saker som inte kan redigeras med Organic Maps
    sw = Eleza hitilafu kwenye ramani au kile ambacho hakiwezi kuhaririwa kwa kutumia Ramani za Kikaboni
    th = อธิบายข้อผิดพลาดในแผนที่หรือสิ่งที่ไม่สามารถแก้ไขได้โดยใช้แผนที่ทั่วไป
    tr = Haritadaki hataları veya Organik Haritalar ile düzenlenemeyen şeyleri tanımlayın
    uk = Опишіть помилки на карті або те, що не можна редагувати за допомогою Organic Maps
    vi = Mô tả lỗi trên bản đồ hoặc những gì không thể chỉnh sửa bằng Bản đồ không phải trả tiền
    zh-Hans = 描述地图上的错误或无法用有机地图编辑的内容
    zh-Hant = 描述地圖中的錯誤或無法使用有機地圖編輯的內容
```

map-per replied:
The shorter version is good. For the 'optional' I prefer the version after the hint in brackets, it stands out more that way.

map-per replied:
The third option "Note to OpenStreetMap volunteers (optional)" is also good, so let's proceed with that one

biodranik commented on data/strings/strings.txt:
```diff
@@ -16542,8 +16542,16 @@
+
+  [editor_about_osm]
+    tags = android
+    en = Your edits are uploaded to the public <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a> database. Please do not add personal or copyrighted information.
```
en = Your edits are uploaded to the public <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a> database. Please do not add personal or copyrighted information.
    af = Jou wysigings word opgelaai na die publieke <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a>-databasis. Moet asseblief nie persoonlike of kopiereginligting byvoeg nie.
    ar = يتم تحميل تعديلاتك على قاعدة البيانات العامة <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">خريطة الشارع المفتوح</a>. يرجى عدم إضافة معلومات شخصية أو محمية بحقوق الطبع والنشر.
    az = Redaktələriniz ictimai <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a> verilənlər bazasına yüklənir. Zəhmət olmasa şəxsi və ya müəllif hüquqları ilə qorunan məlumatları əlavə etməyin.
    be = Вашы праўкі загружаюцца ў публічную базу дадзеных <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a>. Калі ласка, не дадавайце асабістую інфармацыю або інфармацыю, абароненую аўтарскім правам.
    bg = Вашите редакции се качват в публичната база данни <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a>. Моля, не добавяйте лична информация или информация, защитена с авторски права.
    ca = Les vostres edicions es pengen a la base de dades pública <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a>. Si us plau, no afegiu informació personal o amb drets d'autor.
    cs = Vaše úpravy se nahrají do veřejné <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">databáze OpenStreetMap</a>. Nepřidávejte prosím osobní informace nebo informace chráněné autorskými právy.
    da = Dine ændringer uploades til den offentlige <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a>-database. Tilføj venligst ikke personlige eller ophavsretligt beskyttede oplysninger.
    de = Ihre Bearbeitungen werden in die öffentliche <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a> Datenbank hochgeladen. Bitte fügen Sie keine persönlichen oder urheberrechtlich geschützten Informationen hinzu.
    el = Οι επεξεργασίες σας μεταφορτώνονται στη δημόσια βάση δεδομένων <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a>. Παρακαλούμε μην προσθέτετε προσωπικές πληροφορίες ή πληροφορίες που προστατεύονται από πνευματικά δικαιώματα.
    es = Sus ediciones se cargan en la base de datos pública <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a>. Por favor, no añada información personal o protegida por derechos de autor.
    et = Teie muudatused laaditakse üles avalikku <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a> andmebaasi. Palun ärge lisage isiklikku või autoriõigusega kaitstud teavet.
    eu = Zure aldaketak <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a> datu-base publikora kargatzen dira. Mesedez, ez gehitu informazio pertsonalik edo copyrightdun informaziorik.
    fa = ویرایش های شما در پایگاه داده عمومی <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a> آپلود می شود. لطفا اطلاعات شخصی یا دارای حق چاپ را اضافه نکنید.
    fi = Muokkauksesi ladataan julkiseen <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a>-tietokantaan. Älä lisää henkilökohtaisia tai tekijänoikeudella suojattuja tietoja.
    fr = Vos modifications sont téléchargées dans la base de données publique <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a>. Veuillez ne pas ajouter d'informations personnelles ou protégées par le droit d'auteur.
    he = העריכות שלך מועלות למסד הנתונים הציבורי של <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a>. נא לא להוסיף מידע אישי או מוגן בזכויות יוצרים.
    hi = आपके संपादन सार्वजनिक <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a> डेटाबेस पर अपलोड किए जाते हैं। कृपया व्यक्तिगत या कॉपीराइट जानकारी न जोड़ें।
    hu = Az Ön szerkesztései feltöltődnek a nyilvános <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a> adatbázisba. Kérjük, ne adjon hozzá személyes vagy szerzői jogvédelem alatt álló információkat.
    id = Hasil editan Anda akan diunggah ke database <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a> publik. Mohon untuk tidak menambahkan informasi pribadi atau informasi yang memiliki hak cipta.
    it = Le sue modifiche vengono caricate nel database pubblico <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a>. La preghiamo di non aggiungere informazioni personali o protette da copyright.
    ja = あなたの編集は公開されている<a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a>データベースにアップロードされます。個人情報や著作権のある情報は追加しないでください。
    ko = 수정한 내용은 공개 <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a> 데이터베이스에 업로드됩니다. 개인 정보나 저작권이 있는 정보는 추가하지 마세요.
    lt = Jūsų pakeitimai įkeliami į viešą <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a> duomenų bazę. Prašome nedėti asmeninės ar autorių teisėmis apsaugotos informacijos.
    mr = तुमची संपादने सार्वजनिक <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a> डेटाबेसवर अपलोड केली जातात. कृपया वैयक्तिक किंवा कॉपीराइट केलेली माहिती जोडू नका.
    nb = Dine endringer lastes opp til den offentlige <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a>-databasen. Vennligst ikke legg til personlig eller opphavsrettsbeskyttet informasjon.
    nl = Uw bewerkingen worden geüpload naar de openbare <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a> database. Voeg geen persoonlijke of auteursrechtelijk beschermde informatie toe.
    pl = Państwa zmiany są przesyłane do publicznej bazy danych <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a>. Prosimy nie dodawać informacji osobistych lub chronionych prawem autorskim.
    pt = As suas edições são carregadas para a base de dados pública <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a>. Por favor, não adicione informações pessoais ou protegidas por direitos de autor.
    pt-BR = Suas edições são carregadas no banco de dados público <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a>. Não adicione informações pessoais ou protegidas por direitos autorais.
    ro = Edițiile dvs. sunt încărcate în baza de date publică <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a>. Vă rugăm să nu adăugați informații personale sau protejate prin drepturi de autor.
    ru = Ваши правки будут загружены в общедоступную базу данных <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a>. Пожалуйста, не добавляйте личную информацию или информацию, защищенную авторским правом.
    sk = Vaše úpravy sa nahrajú do verejnej databázy <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a>. Prosím, nepridávajte osobné informácie alebo informácie chránené autorskými právami.
    sv = Dina redigeringar laddas upp till den offentliga <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a>-databasen. Lägg inte till personlig eller upphovsrättsskyddad information.
    sw = Mabadiliko yako yanapakiwa kwenye hifadhidata ya <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a> ya umma. Tafadhali usiongeze maelezo ya kibinafsi au yenye hakimiliki.
    th = การแก้ไขของคุณจะถูกอัปโหลดไปยังฐานข้อมูล <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a> สาธารณะ กรุณาอย่าเพิ่มข้อมูลส่วนบุคคลหรือข้อมูลที่มีลิขสิทธิ์
    tr = Düzenlemeleriniz halka açık <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a> veritabanına yüklenir. Lütfen kişisel veya telif hakkıyla korunan bilgiler eklemeyin.
    uk = Ваші правки завантажуються до публічної бази даних <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a>. Будь ласка, не додавайте особисту або захищену авторським правом інформацію.
    vi = Các chỉnh sửa của bạn sẽ được tải lên cơ sở dữ liệu <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a> công khai. Vui lòng không thêm thông tin cá nhân hoặc có bản quyền.
    zh-Hans = 您的编辑内容将上传到公共<a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a>数据库。请勿添加个人或受版权保护的信息。
    zh-Hant = 您的編輯內容將上傳至公用 <a href="https://wiki.openstreetmap.org/wiki/About_OpenStreetMap">OpenStreetMap</a> 資料庫。請不要添加個人或版權資訊。

biodranik commented on data/strings/strings.txt:
```diff
@@ -16498,7 +16498,7 @@
   [editor_other_info]
     tags = android,ios
-    en = Send a note to OSM editors
+    en = Note to OpenStreetMap volunteers
```
en = Note to OpenStreetMap volunteers
    af = Nota aan OpenStreetMap-vrywilligers
    ar = ملاحظة لمتطوعي OpenStreetMap
    az = OpenStreetMap könüllüləri üçün qeyd
    be = Заўвага для валанцёраў OpenStreetMap
    bg = Забележка към доброволците на OpenStreetMap
    ca = Nota per als voluntaris d'OpenStreetMap
    cs = Poznámka pro dobrovolníky OpenStreetMap
    da = Note til OpenStreetMap-frivillige
    de = Hinweis für OpenStreetMap-Freiwillige
    el = Σημείωση για τους εθελοντές του OpenStreetMap
    es = Nota para los voluntarios de OpenStreetMap
    et = Märkus OpenStreetMapi vabatahtlikele
    eu = Oharra OpenStreetMap-eko boluntarioentzat
    fa = به داوطلبان OpenStreetMap توجه داشته باشید
    fi = Huomautus OpenStreetMapin vapaaehtoisille
    fr = Note aux bénévoles d'OpenStreetMap
    he = הערה למתנדבי OpenStreetMap
    hi = OpenStreetMap स्वयंसेवकों के लिए नोट
    hu = Megjegyzés az OpenStreetMap önkénteseinek
    id = Catatan untuk relawan OpenStreetMap
    it = Nota per i volontari di OpenStreetMap
    ja = OpenStreetMapボランティアへの注意
    ko = 오픈스트리트맵 자원 봉사자 참고 사항
    lt = Pastaba "OpenStreetMap" savanoriams
    mr = OpenStreetMap स्वयंसेवकांसाठी टीप
    nb = Merknad til OpenStreetMap-frivillige
    nl = Opmerking voor OpenStreetMap vrijwilligers
    pl = Uwaga dla wolontariuszy OpenStreetMap
    pt = Nota para os voluntários do OpenStreetMap
    ro = Notă pentru voluntarii OpenStreetMap
    ru = Примечание для волонтеров OpenStreetMap
    sk = Poznámka pre dobrovoľníkov OpenStreetMap
    sv = Meddelande till OpenStreetMap-volontärer
    sw = Kumbuka kwa wanaojitolea wa OpenStreetMap
    th = หมายเหตุถึงอาสาสมัคร OpenStreetMap
    tr = OpenStreetMap gönüllülerine not
    uk = До уваги волонтерів OpenStreetMap
    vi = Lưu ý dành cho tình nguyện viên OpenStreetMap
    zh-Hans = OpenStreetMap 志愿者须知
    zh-Hant = OpenStreetMap 志工注意事項

biodranik replied:
Let's check how it looks with new strings (also check the longest translations language).
```
    en = Note to OpenStreetMap volunteers (optional)
    af = Nota aan OpenStreetMap-vrywilligers (opsioneel)
    ar = ملاحظة لمتطوعي OpenStreetMap (اختياري)
    az = OpenStreetMap könüllüləri üçün qeyd (isteğe bağlı)
    be = Заўвага для валанцёраў OpenStreetMap (неабавязкова)
    bg = Забележка към доброволците на OpenStreetMap (по избор)
    ca = Nota per als voluntaris d'OpenStreetMap (opcional)
    cs = Poznámka pro dobrovolníky OpenStreetMap (nepovinné)
    da = Note til OpenStreetMap-frivillige (valgfrit)
    de = Hinweis für OpenStreetMap-Freiwillige (optional)
    el = Σημείωση για τους εθελοντές του OpenStreetMap (προαιρετική)
    es = Nota para los voluntarios de OpenStreetMap (opcional)
    et = Märkus OpenStreetMapi vabatahtlikele (vabatahtlik)
    eu = OpenStreetMap-eko boluntarioentzako oharra (aukerakoa)
    fa = توجه به داوطلبان OpenStreetMap (اختیاری)
    fi = Huomautus OpenStreetMapin vapaaehtoisille (valinnainen)
    fr = Note aux volontaires OpenStreetMap (facultatif)
    he = הערה למתנדבי OpenStreetMap (אופציונלי)
    hi = OpenStreetMap स्वयंसेवकों के लिए नोट (वैकल्पिक)
    hu = Megjegyzés az OpenStreetMap önkénteseinek (nem kötelező)
    id = Catatan untuk relawan OpenStreetMap (opsional)
    it = Nota per i volontari di OpenStreetMap (opzionale)
    ja = OpenStreetMapボランティアへの注意（オプション）
    ko = 오픈스트리트맵 자원 봉사자 참고 사항(선택 사항)
    lt = Pastaba OpenStreetMap savanoriams (neprivaloma)
    mr = OpenStreetMap स्वयंसेवकांसाठी टीप (पर्यायी)
    nb = Merknad til OpenStreetMap-frivillige (valgfritt)
    nl = Opmerking voor OpenStreetMap vrijwilligers (optioneel)
    pl = Uwaga dla wolontariuszy OpenStreetMap (opcjonalnie)
    pt = Nota para os voluntários do OpenStreetMap (opcional)
    ro = Notă pentru voluntarii OpenStreetMap (opțional)
    ru = Примечание для волонтеров OpenStreetMap (необязательно)
    sk = Poznámka pre dobrovoľníkov OpenStreetMap (nepovinné)
    sv = Meddelande till OpenStreetMap-volontärer (valfritt)
    sw = Kumbuka kwa wanaojitolea wa OpenStreetMap (si lazima)
    th = หมายเหตุถึงอาสาสมัคร OpenStreetMap (ไม่บังคับ)
    tr = OpenStreetMap gönüllülerine not (isteğe bağlı)
    uk = Примітка для волонтерів OpenStreetMap (необов'язково)
    vi = Lưu ý dành cho tình nguyện viên OpenStreetMap (tùy chọn)
    zh-Hans = OpenStreetMap 志愿者须知（可选）
    zh-Hant = OpenStreetMap 志工注意事項（可選）
```

map-per replied:
The longest string, stretching over two lines, works as well:
![Screenshot_2024-08-24-20-48-46-520_app organicmaps debug](https://github.com/user-attachments/assets/7f53e796-d528-457c-8f23-9ea807a70dc1)

biodranik reviewed (commented):
Thanks! Will the centered top warning look better than the left-aligned one?

map-per commented:
The top warning indeed looks better when it is centred, so I changed it. Moreover I updated the strings and added translations. The updated screenshots are in the first post (https://github.com/organicmaps/organicmaps/pull/6173#issue-1920818726).

Are we ready to ask the translators for a review?

biodranik commented on data/strings/strings.txt:
```diff
@@ -16542,52 +16542,139 @@
-    vi = Gửi ghi chú cho ban biên tập OSM
-    zh-Hans = 给 OSM 社区发送标记
-    zh-Hant = 給 OSM 社區發送標記
+    en = Note to OpenStreetMap volunteers (optional)
```
@map-per adding `comment = ` with a bit more context would also be helpful.

@organicmaps/translations please help with reviewing these translations. Here is the image with the example of their use:

![выява](https://github.com/user-attachments/assets/c999f6a8-fd9a-4e5e-aa97-71e949d71011)

matheusgomesms commented on data/strings/strings.txt:
```diff
@@ -16542,52 +16542,139 @@
+    nl = Uw bewerkingen worden geüpload naar de openbare <a href="https://wiki.openstreetmap.org/wiki/NL:Wat_is_OpenStreetMap%3F">OpenStreetMap</a> database. Voeg geen persoonlijke of auteursrechtelijk beschermde informatie toe.
+    pl = Państwa zmiany są przesyłane do publicznej bazy danych <a href="https://wiki.openstreetmap.org/wiki/Pl:Wstęp">OpenStreetMap</a>. Prosimy nie dodawać informacji osobistych lub chronionych prawem autorskim.
+    pt = As suas edições são carregadas para a base de dados pública <a href="https://wiki.openstreetmap.org/wiki/Pt:Sobre_o_OpenStreetMap">OpenStreetMap</a>. Por favor, não adicione informações pessoais ou protegidas por direitos de autor.
+    pt-BR = Suas edições são carregadas no banco de dados público <a href="https://wiki.openstreetmap.org/wiki/Pt:Sobre_o_OpenStreetMap">OpenStreetMap</a>. Não adicione informações pessoais ou protegidas por direitos autorais.
```
```suggestion
    pt-BR = As suas edições são enviadas à base de dados pública <a href="https://wiki.openstreetmap.org/wiki/Pt:Sobre_o_OpenStreetMap">OpenStreetMap</a>. Não adicione informações pessoais ou protegidas por direitos autorais.
```

NeatNit commented on data/strings/strings.txt:
```diff
@@ -16542,52 +16542,139 @@
+    fa = خطاهای موجود در نقشه یا مواردی که با استفاده از نقشه های ارگانیک قابل ویرایش نیستند را شرح دهید
+    fi = Kuvaile kartassa olevia virheitä tai asioita, joita ei voi muokata Organic Mapsilla.
+    fr = Décrivez les erreurs sur la carte ou les éléments qui ne peuvent pas être modifiés avec Organic Maps.
+    he = תאר שגיאות במפה או מה שלא ניתן לערוך באמצעות מפות אורגניות
```
```suggestion
    he = תארו שגיאות במפה או דברים שלא ניתן לערוך באמצעות Organic Maps
```

Other auto-translations are good

d4f5409d reviewed (approved):
Hu LGTM

jimcarst reviewed (approved):
nl is OK

map-per commented:
Thank you for the reviews! I updated the translations and added comments

biodranik commented on android/app/src/main/java/app/organicmaps/editor/EditorFragment.java:
```diff
@@ -540,8 +542,6 @@ else if (id == R.id.more_names || id == R.id.show_additional_names)
     else if (id == R.id.add_langs)
       mParent.addLanguage();
-    else if (id == R.id.about_osm)
-      Utils.openUrl(requireActivity(), getString(R.string.osm_wiki_about_url));
```
Is the text/url clickable now?

map-per replied:
Yes, the blue text is clickable. The removed code is form the old separate 'About OpenStreetMap' textview.

biodranik reviewed (approved):
Thanks! @kirylkaveryn can you please check if it's easy to make similar changes on iOS, and if the link in the text is displayed properly/is clickable (or the whole text label is clickable)?

map-per commented:
Can we merge this? The PR doesn't impact iOS, because it uses new string resources, so there is no need to do Android and iOS at the same time.

map-per commented:
:tada:

biodranik commented:
Thanks for your efforts and time, and for reminding about it!

Jean-BaptisteC commented:
@map-per link to wiki osm at the top of the screen in editor not working in other languages (link is not linkified)

map-per commented:
Looks like only the french string has this problem.

```
fr = Vos modifications sont téléchargées dans la base de données publique <a href="https://wiki.openstreetmap.org/wiki/FR:%C3%80_propos_d%E2%80%99OpenStreetMap">OpenStreetMap</a>. Veuillez ne pas ajouter d'informations personnelles ou protégées par le droit d'auteur.

```

Maybe it will work if we replace
"https://wiki.openstreetmap.org/wiki/FR:%C3%80_propos_d%E2%80%99OpenStreetMap"
with 
"https://wiki.openstreetmap.org/wiki/FR:À_propos_d’OpenStreetMap"

map-per commented:
Yes, removing all % encodings from the URLs worked (see  #9291)
